src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs

```diff
@@ -94,6 +99,8 @@ await channel.GetMessagesBeforeAsync(messageList.Last().Id)
                     {
                         fail.Increment(priority);
                     }
+
+                    await _streakService.UpdateStreak(msg.Author.Id);
```

Comment:
Add another argument with the count of completed tasks -- if the count is zero, the streak will also be removed.

Reply:
Also, shouldn't this be only in the first foreach loop? This way it counts every match inside a message... Also, do we have any guarantee that all messages we iterate through will be unique for each user?

Reply:
Another way - I call the method only if I have seen successfully completed task in the message